QuantConnect.DataBento/DataBentoDataProvider.cs

```diff
@@ -113,64 +86,35 @@ public DataBentoProvider(string apiKey)
-                                var resolutionToSubscribe = config.Resolution > Resolution.Tick ? Resolution.Tick : config.Resolution;
-                                var success = _client.Subscribe(config.Symbol, resolutionToSubscribe, config.TickType);
-                                if (success)
-                                {
-                                    Log.Trace($"DataBentoProvider.SubscribeImpl(): Successfully subscribed to {config.Symbol} at {resolutionToSubscribe} resolution");
-                                    
-                                    // Start session after first successful subscription
-                                    lock (_sessionLock)
-                                    {
-                                        if (!_sessionStarted)
-                                        {
-                                            Log.Trace("DataBentoProvider.SubscribeImpl(): Starting DataBento session to receive data");
-                                            _sessionStarted = _client.StartSession();
-                                            if (_sessionStarted)
-                                            {
-                                                Log.Trace("DataBentoProvider.SubscribeImpl(): Session started successfully - data should begin flowing");
-                                            }
-                                            else
-                                            {
-                                                Log.Error("DataBentoProvider.SubscribeImpl(): Failed to start session");
-                                            }
-                                        }
-                                    }
-                                }
-                                else
-                                {
-                                    Log.Error($"DataBentoProvider.SubscribeImpl(): Failed to subscribe to live data for {config.Symbol}");
-                                }
-                            });
-                        }
-                        else
-                        {
-                            Log.Trace($"DataBentoProvider.SubscribeImpl(): Client not connected, skipping subscription for {symbol}");
-                        }
+                    var resolution = config.Resolution > Resolution.Tick ? Resolution.Tick : config.Resolution;
+                    if (!_client.Subscribe(config.Symbol, resolution, config.TickType))
+                    {
+                        Log.Error($"Failed to subscribe to {config.Symbol}");
+                        return false;
                     }
-                    else
+
+                    lock (_sessionLock)
                     {
-                        Log.Trace($"DataBentoProvider.SubscribeImpl(): No config found for {symbol}, skipping");
+                        if (!_sessionStarted)
+                            _sessionStarted = _client.StartSession();
```

Comment:
It is completely wrong.
The logic of 'Start session' shouldn't be here.
Please, put the logic of _client.StartSession() probablly in 
```
public IEnumerator<BaseData>? Subscribe(SubscriptionDataConfig dataConfig, EventHandler newDataAvailableHandler)
```